### DISCUSSION TITLE
Are these instructions still valid for v3?

### DISCUSSION THREAD

**Discussion by frederikhors** [tailwindlabs/tailwindcss#6255] [General]
Not so long ago in the installation instructions of Tailwind it was written to insert the custom css code between `base, components` and `utilities`, like:

```css
@tailwind base;
@tailwind components;

/* MY CUSTOM CSS HERE */

@tailwind utilities;
```

Now I'm using v3 with PostCSS, is that still the case?

Or can I have the following situation instead?

```css

@tailwind base;
@tailwind components;
@tailwind utilities;

/* MY CUSTOM CSS HERE */
```

or

```css
/* MY CUSTOM CSS HERE */

@tailwind base;
@tailwind components;
@tailwind utilities;
```

**Comment by adamwathan** [comment]
You probably want to use the `@layer` directive:

https://tailwindcss.com/docs/functions-and-directives#layer

Where you put your custom CSS depends on what you're trying to do. CSS rules defined later in the stylesheet take precedence over those defined earlier, so if you want to be able to override your CSS with utilities, you want your CSS first. If you want your CSS to take precedence over utilities, you want it last.

This blog post might be helpful:

https://css-tricks.com/precedence-css-order-css-matters/
